packages/connection/src/Manager.php

```diff
@@ -71,8 +71,31 @@ public function is_user_connected( $user_id ) {
 	 * @param Integer $user_id the user identifier.
 	 * @return Object the user object.
 	 */
-	public function get_connected_user_data( $user_id ) {
-		return $user_id;
+	public function get_connected_user_data( $user_id = null ) {
+		if ( ! $user_id ) {
+			$user_id = get_current_user_id();
+		}
+
+		$transient_key = "jetpack_connected_user_data_$user_id";
+
+		if ( $cached_user_data = get_transient( $transient_key ) ) {
+			return $cached_user_data;
+		}
+
+		\Jetpack::load_xml_rpc_client();
```

Comment:
Could we not introduce additional `Jetpack` references to the connection package? I think we should create an IXR client package, and then consume it here.